fixes the failure

```
(awx) bash-4.2$ py.test awx/conf/tests/unit/test_settings.py::test_read_only_setting_assignment
======================================================================================== test session starts ========================================================================================
platform linux -- Python 3.6.6, pytest-3.6.0, py-1.8.0, pluggy-0.6.0
Django settings: awx.settings.development (from ini file)
rootdir: /awx_devel, inifile: pytest.ini
plugins: xdist-1.27.0, timeout-1.3.3, pythonpath-0.7.3, mock-1.10.4, forked-1.0.2, django-3.4.8, cov-2.7.1, celery-4.2.1
collected 1 item                                                                                                                                                                                    

awx/conf/tests/unit/test_settings.py F                                                                                                                                                        [100%]

============================================================================================= FAILURES ==============================================================================================
_________________________________________________________________________________ test_read_only_setting_assignment _________________________________________________________________________________
Traceback (most recent call last):
  File "/awx_devel/awx/conf/tests/unit/test_settings.py", line 290, in test_read_only_setting_assignment
    settings.AWX_SOME_SETTING = 'CHANGED'
  File "/venv/awx/lib64/python3.6/site-packages/django/conf/__init__.py", line 70, in __setattr__
    super(LazySettings, self).__setattr__(name, value)
  File "/venv/awx/lib64/python3.6/site-packages/django/utils/functional.py", line 269, in __setattr__
    setattr(self._wrapped, name, value)
  File "/awx_devel/awx/conf/settings.py", line 484, in __setattr__
    self._set_local(name, value)
  File "/awx_devel/awx/conf/settings.py", line 460, in _set_local
    logger.warning('Attempt to set read only setting "%s".', name)
  File "/usr/lib64/python3.6/logging/__init__.py", line 1319, in warning
    self._log(WARNING, msg, args, **kwargs)
  File "/usr/lib64/python3.6/logging/__init__.py", line 1443, in _log
    self.handle(record)
  File "/usr/lib64/python3.6/logging/__init__.py", line 1453, in handle
    self.callHandlers(record)
  File "/usr/lib64/python3.6/logging/__init__.py", line 1514, in callHandlers
    if record.levelno >= hdlr.level:
  File "/awx_devel/awx/main/utils/handlers.py", line 104, in level
    return logging._nameToLevel[settings.LOG_AGGREGATOR_LEVEL]
  File "/awx_devel/awx/conf/settings.py", line 539, in __getattr_without_cache__
    return getattr(self._wrapped, name)
  File "/awx_devel/awx/conf/settings.py", line 443, in __getattr__
    value = self._get_local(name)
  File "/awx_devel/awx/conf/settings.py", line 359, in _get_local
    self._preload_cache()
  File "/awx_devel/awx/conf/settings.py", line 321, in _preload_cache
    for setting in Setting.objects.filter(key__in=settings_to_cache.keys(), user__isnull=True).order_by('pk'):
  File "/venv/awx/lib64/python3.6/site-packages/django/db/models/query.py", line 250, in __iter__
    self._fetch_all()
  File "/venv/awx/lib64/python3.6/site-packages/django/db/models/query.py", line 1121, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/venv/awx/lib64/python3.6/site-packages/django/db/models/query.py", line 53, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)
  File "/venv/awx/lib64/python3.6/site-packages/django/db/models/sql/compiler.py", line 887, in execute_sql
    cursor = self.connection.cursor()
  File "/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 254, in cursor
    return self._cursor()
  File "/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 229, in _cursor
    self.ensure_connection()
  File "/venv/awx/lib64/python3.6/site-packages/pytest_django/plugin.py", line 749, in _blocking_wrapper
    "Database access not allowed, "
  File "/venv/awx/lib64/python3.6/site-packages/_pytest/outcomes.py", line 92, in fail
    raise Failed(msg=msg, pytrace=pytrace)
Failed: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.
--------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------
2019-05-22 12:55:09,846 WARNING  awx.conf.settings Attempt to set read only setting "AWX_SOME_SETTING".
===================================================================================== 1 failed in 0.18 seconds ======================================================================================
```

this would only happen if you make memcache values stale first.

--- a/awx/main/utils/filters.py
+++ b/awx/main/utils/filters.py
@@ -18,7 +18,7 @@ from django.conf import settings
 from awx.main.constants import LOGGER_BLACKLIST
 from awx.main.utils.common import get_search_fields
 
-__all__ = ['SmartFilter', 'ExternalLoggerEnabled']
+__all__ = ['SmartFilter', 'ExternalLoggerEnabled', 'DynamicLevelFilter']
 
 logger = logging.getLogger('awx.main.utils')
 
@@ -48,9 +48,18 @@ class FieldFromSettings(object):
             instance.settings_override[self.setting_name] = value
 
 
+def record_is_blacklisted(record):
+    """Given a log record, return True if it is considered to be in
+    the logging blacklist, return False if not
+    """
+    for logger_name in LOGGER_BLACKLIST:
+        if record.name.startswith(logger_name):
+            return True
+    return False
+
+
 class ExternalLoggerEnabled(Filter):
 
-    lvl = FieldFromSettings('LOG_AGGREGATOR_LEVEL')
     enabled_loggers = FieldFromSettings('LOG_AGGREGATOR_LOGGERS')
     enabled_flag = FieldFromSettings('LOG_AGGREGATOR_ENABLED')
 
@@ -64,18 +73,16 @@ class ExternalLoggerEnabled(Filter):
             setattr(self, field_name, field_value)
 
     def filter(self, record):
-        """
-        Uses the database settings to determine if the current
-        external log configuration says that this particular record
-        should be sent to the external log aggregator
+        """Filters out all log records if LOG_AGGREGATOR_ENABLED is False
+        or if the particular logger is not in LOG_AGGREGATOR_LOGGERS
+        should only be used for the external logger
 
         False - should not be logged
         True - should be logged
         """
-        # Logger exceptions
-        for logger_name in LOGGER_BLACKLIST:
-            if record.name.startswith(logger_name):
-                return False
+        # Do not send exceptions to external logger
+        if record_is_blacklisted(record):
+            return False
         # General enablement
         if not self.enabled_flag:
             return False
@@ -93,6 +100,24 @@ class ExternalLoggerEnabled(Filter):
             else:
                 base_name = record.name
             return bool(base_name in loggers)
+
+
+class DynamicLevelFilter(Filter):
+
+    def filter(self, record):
+        """Filters out logs that have a level below the threshold defined
+        by the databse setting LOG_AGGREGATOR_LEVEL
+        """
+        if record_is_blacklisted(record):
+            # Fine to write blacklisted loggers to file, apply default filtering level
+            cutoff_level = logging.WARNING
+        else:
+            try:
+                cutoff_level = logging._nameToLevel[settings.LOG_AGGREGATOR_LEVEL]
+            except Exception:
+                cutoff_level = logging.WARNING
+
+        return bool(record.levelno >= cutoff_level)
 
 
 def string_to_type(t):

--- a/awx/main/utils/handlers.py
+++ b/awx/main/utils/handlers.py
@@ -19,12 +19,11 @@ from django.conf import settings
 from requests_futures.sessions import FuturesSession
 
 # AWX
-from awx.main.constants import LOGGER_BLACKLIST
 from awx.main.utils.formatters import LogstashFormatter
 
 
 __all__ = ['BaseHTTPSHandler', 'TCPHandler', 'UDPHandler',
-           'AWXProxyHandler', 'RotatingProductionLogHandler']
+           'AWXProxyHandler']
 
 
 logger = logging.getLogger('awx.main.utils.handlers')
@@ -90,28 +89,6 @@ class SocketResult:
 
     def result(self):
         return self
-
-
-class DynamicLevelMixin(object):
-
-    @property
-    def level(self):
-        from django.conf import settings
-        for logger_name in LOGGER_BLACKLIST:
-            if self.name.startswith(logger_name):
-                return 30  # WARNING
-        try:
-            return logging._nameToLevel[settings.LOG_AGGREGATOR_LEVEL]
-        except Exception:
-            return 30  # WARNING
-
-    @level.setter
-    def level(self, level):
-        pass  # no-op, this value comes from the database
-
-
-class RotatingProductionLogHandler(logging.handlers.RotatingFileHandler, DynamicLevelMixin):
-    pass
 
 
 class BaseHandler(logging.Handler):
@@ -295,7 +272,7 @@ HANDLER_MAPPING = {
 }
 
 
-class AWXProxyHandler(logging.Handler, DynamicLevelMixin):
+class AWXProxyHandler(logging.Handler):
     '''
     Handler specific to the AWX external logging feature
 

--- a/awx/settings/defaults.py
+++ b/awx/settings/defaults.py
@@ -966,6 +966,9 @@ LOGGING = {
         'external_log_enabled': {
             '()': 'awx.main.utils.filters.ExternalLoggerEnabled'
         },
+        'dynamic_level_filter': {
+            '()': 'awx.main.utils.filters.DynamicLevelFilter'
+        }
     },
     'formatters': {
         'simple': {
@@ -1005,7 +1008,7 @@ LOGGING = {
         'external_logger': {
             'class': 'awx.main.utils.handlers.AWXProxyHandler',
             'formatter': 'json',
-            'filters': ['external_log_enabled'],
+            'filters': ['external_log_enabled', 'dynamic_level_filter'],
         },
         'mail_admins': {
             'level': 'ERROR',
@@ -1014,8 +1017,8 @@ LOGGING = {
         },
         'tower_warnings': {
             # don't define a level here, it's set by settings.LOG_AGGREGATOR_LEVEL
-            'class':'awx.main.utils.handlers.RotatingProductionLogHandler',
-            'filters': ['require_debug_false'],
+            'class': 'logging.handlers.RotatingFileHandler',
+            'filters': ['require_debug_false', 'dynamic_level_filter'],
             'filename': os.path.join(LOG_ROOT, 'tower.log'),
             'maxBytes': 1024 * 1024 * 5, # 5 MB
             'backupCount': 5,
@@ -1023,8 +1026,8 @@ LOGGING = {
         },
         'callback_receiver': {
             # don't define a level here, it's set by settings.LOG_AGGREGATOR_LEVEL
-            'class':'awx.main.utils.handlers.RotatingProductionLogHandler',
-            'filters': ['require_debug_false'],
+            'class': 'logging.handlers.RotatingFileHandler',
+            'filters': ['require_debug_false', 'dynamic_level_filter'],
             'filename': os.path.join(LOG_ROOT, 'callback_receiver.log'),
             'maxBytes': 1024 * 1024 * 5, # 5 MB
             'backupCount': 5,
@@ -1032,8 +1035,8 @@ LOGGING = {
         },
         'dispatcher': {
             # don't define a level here, it's set by settings.LOG_AGGREGATOR_LEVEL
-            'class':'awx.main.utils.handlers.RotatingProductionLogHandler',
-            'filters': ['require_debug_false'],
+            'class': 'logging.handlers.RotatingFileHandler',
+            'filters': ['require_debug_false', 'dynamic_level_filter'],
             'filename': os.path.join(LOG_ROOT, 'dispatcher.log'),
             'maxBytes': 1024 * 1024 * 5, # 5 MB
             'backupCount': 5,
@@ -1050,8 +1053,8 @@ LOGGING = {
         },
         'task_system': {
             # don't define a level here, it's set by settings.LOG_AGGREGATOR_LEVEL
-            'class':'awx.main.utils.handlers.RotatingProductionLogHandler',
-            'filters': ['require_debug_false'],
+            'class': 'logging.handlers.RotatingFileHandler',
+            'filters': ['require_debug_false', 'dynamic_level_filter'],
             'filename': os.path.join(LOG_ROOT, 'task_system.log'),
             'maxBytes': 1024 * 1024 * 5, # 5 MB
             'backupCount': 5,

